File: backend/app/api/routes/ratings.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session

from app.api.deps import get_current_user, get_db, require_role
from app.models.match import Match
from app.models.match_player_rating import MatchPlayerRating
from app.models.user import User
from app.schemas.rating import RatingOut, RatingsSaveRequest

router = APIRouter(prefix="/matches", tags=["ratings"])


@router.get("/{match_id}/ratings", response_model=list[RatingOut])
def list_ratings(
    match_id: int,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    match = (
        db.query(Match)
        .filter(Match.id == match_id, Match.club_id == current_user.club_id)
        .first()
    )
    if not match:
        raise HTTPException(status_code=404, detail="Match not found")

    return (
        db.query(MatchPlayerRating)
        .filter(MatchPlayerRating.match_id == match_id)
        .order_by(MatchPlayerRating.player_id.asc())
        .all()
    )


@router.put("/{match_id}/ratings", response_model=list[RatingOut])
def save_ratings(
    match_id: int,
    payload: RatingsSaveRequest,
    db: Session = Depends(get_db),
    current_user: User = Depends(require_role("coach")),
):
    from app.exceptions import AppError
    from app.services.ratings_service import save_ratings as ratings_save

    try:
        return ratings_save(db, match_id, current_user, payload)
    except AppError as e:
        raise HTTPException(status_code=e.status_code, detail=e.message)

File: frontend/src/api/ratings.ts
import { api } from "./client";

export type RatingRow = {
  id: number;
  match_id: number;
  user_id: number;
  player_id: number;
  rating: number;
  note?: string | null;
};

export async function listRatings(matchId: number) {
  const res = await api.get<RatingRow[]>(`/matches/${matchId}/ratings`);
  return res.data;
}

export async function saveRatings(matchId: number, items: Array<{ player_id: number; rating: number; note?: string | null }>) {
  const res = await api.put<RatingRow[]>(`/matches/${matchId}/ratings`, { items });
  return res.data;
}

